### PR TITLE
Mongo storage update

### PR DESCRIPTION
**Pull request by max-vasin**
* mongo storage: fixed bug with missing timestamp boundary events ($gt -> $gte)
* mongo storage: events loading algorithm changed to reduce memory pressure
* mongo storage: native mongo tests are added

### CODE REVIEW

File: packages/storage-adapters/resolve-storage-mongo/package.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
   "devDependencies": {
     "chai": "^3.5.0",
     "jest": "^22.1.4",
-    "sinon": "^3.3.0"
+    "sinon": "^3.3.0",
+    "bluebird": "^2.9.34",
```
Comment (TatyanaRyzh) [inline packages/storage-adapters/resolve-storage-mongo/package.json]: Please, clarify, why you add the 'bluebird' package? Is it a syntax sugar or really necessary thing?

**Comment by max-vasin** [comment]
Wrong git flow.

File: packages/storage-adapters/resolve-storage-mongo/test/index.tests.js [hunk 2]
```diff
@@ -5,15 +5,15 @@ import { ConcurrentError } from 'resolve-storage-base'
...
   const sinon = require('sinon')
-  let foundArray = []
+  let foundCursor = { current: 0, items: [] }
   let isRejectInsert = false
 
   return {
     _setFindResult: array => {
       if (Array.isArray(array)) {
-        foundArray = array
+        foundCursor = { current: 0, items: array }
```
Comment (TatyanaRyzh) [inline packages/storage-adapters/resolve-storage-mongo/test/index.tests.js]: foundCursor = { current: 0, items: Array.isArray(array) ? array : [] }

Comment (superroma) [reply packages/storage-adapters/resolve-storage-mongo/test/index.tests.js]: Check two times to make sure?

**Pull request closed without merging**